parser/src/hydrate.ts

```diff
@@ -70,6 +78,27 @@ export async function hydrateTreeWithAuthorship(
       parent = (await parseCommitLight(repo, childCommit.parent)).parent
     }
   }
+  discardContentField(hydratedCommit.tree)
 
   return hydratedCommit
 }
+
+function addAuthorsField(tree: HydratedGitTreeObject) {
```

Comment:
Can we when possible please avoid side-effect based functions like this 🥺
This is fine though, but i think our code will become messy and highly coupled if we do it too much

Reply:
Yes, the problem is that the current hydration approach is based on this. If this was f#, it would maybe be easier to write more functional code.